x/erc20/keeper/genesis.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/evmos/v16/x/erc20/types"
+	"github.com/evmos/evmos/v16/x/evm/statedb"
+)
+
+// SetGenesisTokenPairs stores in state the provided token pairs in genesis
+func (k Keeper) SetGenesisTokenPairs(ctx sdk.Context, pairs []types.TokenPair) error {
+	for i, pair := range pairs {
+		contractAddr := pair.GetERC20Contract()
+		coinMeta, err := k.getGenesisTokenPairMeta(ctx, pair, i)
+		if err != nil {
+			return fmt.Errorf("error while generating metadata for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB := statedb.New(ctx, k.evmKeeper, statedb.TxConfig{})
+		code, err := k.generateContractCode(ctx, stateDB, coinMeta, contractAddr)
+		if err != nil {
+			return fmt.Errorf("error while getting contract code for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB.SetCode(contractAddr, code)
+
+		if err := stateDB.Commit(); err != nil {
+			return fmt.Errorf("error while storing contract for genesis pair denom %s. %w", pair.Denom, err)
+		}
+
+		id := pair.GetID()
+		k.SetTokenPair(ctx, pair)
+		k.SetDenomMap(ctx, pair.Denom, id)
+		k.SetERC20Map(ctx, pair.GetERC20Contract(), id)
+	}
+
+	return nil
+}
+
+// getGenesisTokenPairMeta is a helper function to generate token pair metadata for the genesis token pairs
```

Comment:
A genesis token pair is a token pair that will be registered from genesis. A current use case of this would be for outposts e2e tests, where we need to register a token pair to make it work, and is not possible to do it otherwise because the RegisterCoinProposal was deprecated.

Same will apply when the STR v2 is implemented. We'll need to have the option to start a chain from genesis with some token pairs registered to cover test cases.

Reply:
please update the comments with the info